### PR TITLE
implement custom data loader

### PR DESCRIPTION
**Pull request by chrisantonellis**
* implement logic for loading and executing a custom data loader
* implemented associated tests

### CODE REVIEW

File: little_cheesemonger/_cli.py [hunk 1]
```diff
@@ -16,13 +16,13 @@
 
 @click.command()
 @click.argument("directory", type=Path, default=Path("."))
-@click.option("-l", "--loader")
+@click.option("-l", "--loader", "loader_import_path")
```
Comment (chrisantonellis) [inline little_cheesemonger/_cli.py:19]: argument name cleanup for UX purposes

File: tests/test_cli.py [hunk 2]
```diff
@@ -13,82 +13,103 @@ def cli_runner():
...
 def test_process_kwargs__malformed_mapping_string__raise_LittleCheesemongerError():
     with pytest.raises(LittleCheesemongerError):
-        _process_kwargs(("invalid",))
+        process_kwargs(("invalid",))
 
 
-def test_entrypoint__no_args(cli_runner):
-    result = cli_runner.invoke(entrypoint)
+def test_entrypoint__no_args(cli_runner, default_loader):
+    result = cli_runner.invoke(entrypoint, catch_exceptions=False)
```
Comment (chrisantonellis) [inline tests/test_cli.py:45]: `catch_exceptions` is required here to make the test not swallow unhandled exceptions

File: tests/test_cli.py [hunk 3]
```diff
@@ -13,82 +13,103 @@ def cli_runner():
 
 
 @pytest.fixture(autouse=True)
+def log_level(caplog):
+    caplog.set_level(logging.DEBUG)
```
Comment (fhightower) [inline tests/test_cli.py:17]: This is a helpful snippet! I'll have to use this one elsewhere.

**Review by fhightower — APPROVED** [review]
Except for the CI failures, this looks good to me! I appreciated your review comments too; those are really helpful and something I'll have to do as well.

**Pull request merged**